sklearn/metrics/pairwise.py

```diff
@@ -720,6 +720,72 @@ def haversine_distances(X, Y=None):
     return DistanceMetric.get_metric('haversine').pairwise(X, Y)
 
 
+def pairwise_wasserstein_distances(X, Y, X_d, Y_d):
+    """Computes the pairwise wasserstein distances of two groups of
+    data points with uncertainties
+    wikipedia: https://en.wikipedia.org/wiki/Wasserstein_metric
```

Comment:
We usually document details in the User Guide, in this case `doc/modules/metrics.rst`. Apart from anything else, Wikipedia can't be trusted to remain aligned with our description.

It's worth noting that this implementation does not consider covariances.

Reply:
thanks for the comment. I will add description to doc/modules/metrics.rst